### PR TITLE
Add Auth service which calls backend to login

### PR DESCRIPTION
**Pull request by nakhbari**
Issue: #752 

- [x] Create Auth service that handles server login
- [x] Store JWT in local storage

### CODE REVIEW

File: web/app/services/auth.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {inject, TestBed} from '@angular/core/testing';
+import {mockLoginResponse} from '../common/test_helpers/mock_login_data';
+import {AuthService, LoginRequest, LoginResponse} from './auth.service';
+
+const FAKE_LOGIN_REQUEST: LoginRequest = {
+  username: 'tacoRocat',
```
Comment (minuscorp) [inline web/app/services/auth.service.spec.ts]: i think the parameter is `email`, but I can change it in the backend. Also, how are you sending this parameters? (`json/application`, `form-data`?)

File: web/app/services/auth.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    mockLocalStorage.clear();
+  });
+
+  it('should should make login request to backend', () => {
+    let loginResponse: LoginResponse;
+    authService.login(FAKE_LOGIN_REQUEST).subscribe((response) => {
+      loginResponse = response;
+    });
+
+    const loginRequest = mockHttp.expectOne('/data/login');
```
Comment (minuscorp) [inline web/app/services/auth.service.spec.ts]: it should be just  `/login` (?)

File: web/app/services/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import {HttpClient} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import * as moment from 'moment';
+import {Observable} from 'rxjs/Observable';
+import {shareReplay, tap} from 'rxjs/operators';
+import {LocalStorageKeys} from '../common/constants';
+
+// Auth server is currently locally hosted.
+const HOSTNAME = '/data';
```
Comment (minuscorp) [inline web/app/services/auth.service.ts]: Yeah, in the backend is not under `/data` path, is just `/login`

File: web/app/services/auth.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    expect(loginRequest.request.body).toBe(FAKE_LOGIN_REQUEST);
+    loginRequest.flush(mockLoginResponse);
+
+    expect(loginResponse.token).toBe('12345');
+  });
+
+  it('should store the token in local storage', () => {
+    authService.login(FAKE_LOGIN_REQUEST).subscribe();
+
+    const loginRequest = mockHttp.expectOne('/data/login');
```
Comment (minuscorp) [inline web/app/services/auth.service.spec.ts]: dito

File: web/app/services/auth.service.spec.ts (hunk 2 above)
Comment (minuscorp) [reply web/app/services/auth.service.spec.ts]: So we should expose the whole json API under the `/api` prefix?

Comment (nakhbari) [reply web/app/services/auth.service.spec.ts]: I'm good with that. Let's start with `/login`. Leave the rest in `/data` for now

File: web/app/services/auth.service.spec.ts (hunk 1 above)
Comment (nakhbari) [reply web/app/services/auth.service.spec.ts]: this would be a part of the HTTP request's body.

File: web/app/services/auth.service.spec.ts (hunk 2 above)
Comment (nakhbari) [reply web/app/services/auth.service.spec.ts]: We can change this in the front-end once the backend is ready

**Pull request merged**